pandas/tests/io/parser/common.py

```diff
@@ -1695,3 +1695,40 @@ class InvalidBuffer(object):
 
             with tm.assertRaisesRegexp(ValueError, msg):
                 self.read_csv(mock.Mock())
+
+    def test_skip_bad_lines(self):
+        data = 'a\n1\n1,2,3\n4\n5,6,7'
+
+        with tm.assertRaises(ParserError):
+            self.read_csv(StringIO(data))
+
+        with tm.assertRaises(ParserError):
+            self.read_csv(StringIO(data), error_bad_lines=True)
+
+        stderr = sys.stderr
+        expected = DataFrame({'a': [1, 4]})
```

Comment:
I actually took this setup from elsewhere in the code.  You are correct that is a `pytest` facility of doing this (see <a href="https://docs.pytest.org/en/latest/capture.html">here</a>).  However, if I'm going to make that change, I'd rather apply it to all places where we do this `stderr = sys.stderr`.  

Could this also be in a follow-up?